src/main/java/org/folio/circulation/support/Result.java

```diff
@@ -58,6 +62,14 @@
     return firstResult.combine(secondResult, combiner);
   }
 
+  static <T> Result<List<T>> combineAll(List<Result<T>> results) {
```

Comment:
This can be replaced with something like below, to make invocation easier and to remove the need for the extra dependency (I've tried variants of this signature with varargs or streams as well)
```
  static <T> Result<List<T>> combineAll(
    List<Result<T>> results1, List<Result<T>> results2) {

    return Stream.concat(results1.stream(), results2.stream())
      .map(r -> r.map(Stream::of))
      .reduce(of(Stream::empty), Result::combineResultStream)
      .map(stream -> stream.collect(Collectors.toList()));
  }

  static <T> Result<Stream<T>> combineResultStream(
    Result<Stream<T>> results1, Result<Stream<T>> results2) {

    return Result.combine(results1, results2, Stream::concat);
  }
```

Reply:
Thanks for the suggestion, I like this more. The only thing I've changed is parameter list:
```
 static <T> Result<List<T>> combineAll(
    List<Result<T>>... results) {

    return Stream.of(results).flatMap(Collection::stream)
      .map(r -> r.map(Stream::of))
      .reduce(of(Stream::empty), Result::combineResultStream)
      .map(stream -> stream.collect(Collectors.toList()));
  }
```
It allows to use this method with only one list (I actually need this in other place)

Reply:
Sure, I tried that, and wasn't sure given the warning about potentially unsafe var args.

Thank you for adding Java doc.